File: Classes/EVEOnlineAPI/Map/EVEJumps.m
//
//  EVEJumps.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 8/20/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEJumps.h"


@implementation EVEJumpsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"solarSystemID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"shipJumps":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	
	return scheme;
}

@end


@implementation EVEJumps

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"solarSystems":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEJumpsItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEJumps.swift
//
//  EVEJumps.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEJumpsItem: EVEObject {
	public var solarSystemID: Int = 0
	public var shipJumps: Int = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"solarSystemID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"shipJumps":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		]
	}
}

public class EVEJumps: EVEResult {
	public var solarSystems: [EVEJumpsItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"solarSystems":EVESchemeElementType.Rowset(elementName: nil, type: EVEJumpsItem.self, transformer: nil)
		]
	}
}

